1-js/06-advanced-functions/03-closure/10-make-army/solution.md

```diff
@@ -51,24 +51,29 @@ function makeArmy() {
 }
 ```
 
-...We can see that it lives in the lexical environment associated with the current `makeArmy()` run. But when `army[5]()` is called, `makeArmy` has already finished its job, and `i` has the last value: `10` (the end of `while`).
+ we can see that it lives in the lexical environment associated with the current `makeArmy()` run. But when `army[5]()` is called, `makeArmy` has already finished its job, and the final value of `i` is `10` (at the end of `while`).
 
-As a result, all `shooter` functions get from the outer lexical envrironment the same, last value `i=10`.
+As a result, all `shooter` functions get the same value from the outer lexical environment and that is, the last value, `i=10`.
 
-We can fix it by moving the variable definition into the loop:
+![](task-while-lexenv-makearmy.svg)
 
-```js run demo
-function makeArmy() {
+As you can see above, on each iteration of a `while {...} ` block, a new lexical environment is created. This implies that as long as we store the value of `i` in a variable in the `while {...}` block, created Lexical Environment will have that variable with value of `i`.
```

Comment:
The second phrase is hard to understand.

Reply:
`created Lexical Environment will have that variable with value of `i`.`

^This?

Reply:
Does the entire sentence not making any sense or is it the way it is phrased? Do you have any idea how to simplify it? I couldn't think of anything atm. :-(